### PR TITLE
RN-571 

### PR DESCRIPTION
**Pull request by aivytran**
[RN-571](https://rhumbix.atlassian.net/browse/RN-571)

Currently, mobile expects and stores number and integer types as string. That's why after transform, the hours, which are number, does not render properly on mobile. Same bug if user create T&M on web with number/integer fields. 

This is because ` getTransformer` function in `Textbox` ([here](https://github.com/Rhumbix/tcomb-form-native/blob/master/lib/components.js#L215)) in `tcomb-form-native` is comparing `this.typeInfo.innerType === t.Number` which will return `False` for number/integer. This comparison is supposed to return `True` to get the right format/parse functions for number/integer.

It doesn't return properly because in `tcomb-json-schema`, we changes type `number` from `t.Number` to `util.Num` ([here](https://github.com/Rhumbix/tcomb-json-schema/commit/f5e136ed4e150921bd88d9d62a3a0059658daf80) )

So I updated `tcomb-form-native` to compare type with `util.Num`/`util.Int` instead of `t.Number`

### CODE REVIEW

File: lib/components.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ import {
   getTypeFromUnion,
   getComponentOptions
 } from './util';
-
+import { util } from "@rhumbix/tcomb-json-schema"
```
Comment (CarolynWebster) [inline lib/components.js]: I don't see `@rhumbix/tcomb-json-schema` in the `package.json`, so I imagine we would need that in order to import that here, right? I made that change to `util.Num` in `tcomb-json-schema` bc of the issue with numbers being strings failing validation, but I am wondering if we are fixing the wrong issue here and should fix the root issue that Number fields are returning Strings instead of Numbers.

Comment (tclancy) [reply lib/components.js]: Same. Shouldn't this be forcing the values rather than just testing the type? And why the need to also check against `util.Int`? Doesn't `util.Number` handle that? This needs tests as well to ensure it works and see how it handles empty strings.

Comment (aivytran) [reply lib/components.js]: Testing the type properly there will return the right format and parse functions for number ([here](https://github.com/Rhumbix/tcomb-form-native/blob/master/lib/components.js#L288)). That will fix the numbers being stored as string issue. The `format` function will handle null value and format number into string to show it on textbox UI properly. When save, the `parse` function with parse string back to number to store. So type `number` will be stored as numbers instead of string.

[`util.Int` and `util.Num`](https://github.com/Rhumbix/tcomb-json-schema/blob/master/src/util.js#L17) have different `name` and `predicate` so they need to be checked separately.

Comment (aivytran) [reply lib/components.js]: I will add some tests for this on RN repo. It's easier to add test there.

Comment (CarolynWebster) [reply lib/components.js]: If it is possible to get the test in this repo that would be best since this is where the functionality lives, and we shouldn't be writing tests for outside libraries in our RN/WA repos. It does appear there is a tests folder in this repo, so might be able to derive from there what you would need to do?

Comment (aivytran) [reply lib/components.js]: hmm, yea, this repo has unit tests using Tape. I haven't used Tape before. But i will add couple tests for this.

File: test/index.js [hunk 2]
```diff
@@ -178,7 +179,7 @@ test('Textbox:help', function (tape) {
 });
 
 test('Textbox:value', function (tape) {
-  tape.plan(3);
+  tape.plan(6);
```
Comment (tclancy) [inline test/index.js:182]: Please add a comment explaining what this magic number does or turn it into a variable that explains it.

File: package.json [hunk 3]
```diff
@@ -23,16 +23,22 @@
...
   "dependencies": {
-    "tcomb-validation": "^3.0.0",
-    "@rhumbix/tcomb-json-schema": "1.0.6"
+    "@rhumbix/tcomb-json-schema": "1.0.6",
+    "babel-eslint": "^10.0.3",
+    "tcomb-validation": "^3.0.0"
   },
   "devDependencies": {
-    "@babel/core": "^7.8.4",
-    "@babel/preset-env": "^7.8.4",
```
Comment (tclancy) [inline package.json]: Is there a reason to revert/ remove these?

File: package.json [hunk 4]
```diff
@@ -23,16 +23,22 @@
...
+    "babel": "5.8.34",
+    "babel-core": "5.8.34",
+    "eslint": "^4.19.1",
+    "eslint-config-prettier": "^2.9.0",
+    "eslint-plugin-prettier": "^2.6.0",
+    "eslint-plugin-react": "2.5.2",
+    "prettier": "^1.7.4",
+    "prop-types": "^15.5.10",
+    "react": "^15.6.1",
     "tape": "3.5.0"
```
Comment (livcarman) [inline package.json]: Where are all these changes coming from? For example, the build command for this package uses rollup, so why is rollup being uninstalled?

File: package.json [hunk 5]
```diff
@@ -23,16 +23,22 @@
...
+    "@types/prettier": "^1.7.0",
+    "babel": "5.8.34",
+    "babel-core": "5.8.34",
+    "eslint": "^4.19.1",
+    "eslint-config-prettier": "^2.9.0",
+    "eslint-plugin-prettier": "^2.6.0",
+    "eslint-plugin-react": "2.5.2",
+    "prettier": "^1.7.4",
+    "prop-types": "^15.5.10",
+    "react": "^15.6.1",
```
Comment (tclancy) [inline package.json]: Should this be in `dependencies` and `babel-eslint` in `devDependencies`? I can't imagine you need the TypeScript dependencies at all, do you?

File: test/index.js (hunk 2 above)
Comment (aivytran) [reply test/index.js:182]: The number is just to specify how many assertions we are going to run in the test. tape has `.plan` at the beginning of every test.

**Comment by aivytran** [comment]
Sorry for the mixed up. Pushed a wrong version 😅. Just need to add couple packages for the tests.

File: test/index.js (hunk 2 above)
Comment (tclancy) [reply test/index.js:182]: [Is it strictly necessary?}(https://github.com/substack/tape#tplann) The docs don't make it clear at a glance. Looks like there's a conflict to be resolved in the PR.

Comment (aivytran) [reply test/index.js:182]: Yea, it is. If we don't declare the plan, it will error out. 

I resolved the conflict. One question, I saw that we removed all the rollup packages on master. Liv mentioned below that the build command for this package uses rollup. I assume that we made some changes?

Comment (tclancy) [reply test/index.js:182]: Yes, I reverted those non-working changes as they were in Carolyn's way. Should be fine to leave them out now.

Comment (aivytran) [reply test/index.js:182]: Oh ok, makes sense.

**Comment by aivytran** [comment]
@tclancy @CarolynWebster  Thanks for all the comments -- everything should be resolved now. Could you guys take another look and let me know if this is GTG?

**Review by tclancy — APPROVED** [review]

**Comment by forrestj** [comment]
@jongoncalves  Hi, can we merge this one?

**Comment by aivytran** [comment]
Merged!
